### PR TITLE
expression: change the String() method of `expression.Column`

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve? 

Not rely on `*Name`s.
To remove `*Name`s from `expression.Column`.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

### CODE REVIEW

**Comment by winoros** [comment]
This pr needs to modify the result of mysql-test and randgen-test

File: bindinfo/bind_test.go [hunk 1]
```diff
@@ -328,39 +328,39 @@ func (s *testSuite) TestGlobalAndSessionBindingBothExist(c *C) {
 	tk.MustExec("create table t2(id int)")
 
 	tk.MustQuery("explain SELECT * from t1,t2 where t1.id = t2.id").Check(testkit.Rows(
-		"HashLeftJoin_8 12487.50 root inner join, inner:TableReader_15, equal:[eq(test.t1.id, test.t2.id)]",
+		"HashLeftJoin_8 12487.50 root inner join, inner:TableReader_15, equal:[eq(Column#1, Column#3)]",
```
Comment (lamxTyler) [inline bindinfo/bind_test.go]: It is hard to read the result now. Is there no easy way to keep the origin format?

Comment (winoros) [reply bindinfo/bind_test.go]: We can create the map that recording the column's changing. But it's hard to keep the original result.

**Comment by winoros** [comment]
We have some ways to make it more readable. But this pr is large enough.
I want to fix it later.

**Pull request merged**